packages/react/src/components/Callout/CalloutContent.base.tsx

```diff
@@ -198,70 +202,81 @@ function usePositions(
+  const [elementPositions, setElementPositions] = React.useState<ICalloutPositionedInfo>();
   const async = useAsync();
-  const { hidden, target, finalHeight, onPositioned, directionalHint } = props;
 
-  React.useEffect(() => {
-    if (!hidden) {
-      const timerId = async.requestAnimationFrame(() => {
-        // If we expect a target element to position against, we need to wait until `targetRef.current`
-        // is resolved. Otherwise we can try to position.
-        const expectsTarget = !!target;
-
-        if (hostElement.current && calloutElement.current && (!expectsTarget || targetRef.current)) {
-          const currentProps: IPositionProps = {
-            ...props,
-            target: targetRef.current!,
-            bounds: getBounds(),
-          };
-          // If there is a finalHeight given then we assume that the user knows and will handle
-          // additional positioning adjustments so we should call positionCard
-          const newPositions: ICalloutPositionedInfo = finalHeight
-            ? positionCard(currentProps, hostElement.current, calloutElement.current, positions)
-            : positionCallout(currentProps, hostElement.current, calloutElement.current, positions);
-
-          // Set the new position only when the positions are not exists or one of the new callout positions
-          // are different. The position should not change if the position is within 2 decimal places.
-          if (
-            (!positions && newPositions) ||
-            (positions && newPositions && !arePositionsEqual(positions, newPositions) && positionAttempts.current < 5)
-          ) {
-            // We should not reposition the callout more than a few times, if it is then the content is likely resizing
-            // and we should stop trying to reposition to prevent a stack overflow.
-            positionAttempts.current++;
-            setPositions(newPositions);
-          } else if (positionAttempts.current > 0) {
-            // Only call the onPositioned callback if the callout has been re-positioned at least once.
-            positionAttempts.current = 0;
-            onPositioned?.(positions);
-          }
+  /**
+   * Sets the current position of Callout upon resize. Callout will resize when:
+   *
+   * 1. `elementPositions` is undefined (initial render)
+   * 2. `elementPositions` does not equal the `newElementPositions`
+   * 3. There is a two decimal place difference in position.
+   *
+   * @param newElementPositions The incoming positions for the Callout
+   */
+  const setCalloutPositions = React.useCallback(
+    (newElementPositions: ICalloutPositionedInfo) => {
+      if (
+        (elementPositions === undefined && newElementPositions) ||
+        (elementPositions && newElementPositions && !arePositionsEqual(elementPositions, newElementPositions))
+      ) {
+        if (elementPositions !== undefined) {
+          onPositioned?.(elementPositions);
```

Comment:
Should you be passing `newElementPositions` here?
```suggestion
          onPositioned?.(newElementPositions);
```

Reply:
It should be newElementPositions